test/tstDetailsAlgorithms.cpp

```diff
@@ -56,8 +56,8 @@ BOOST_AUTO_TEST_CASE(intersects)
   // uninitialized box does not even intersect with itself
   STATIC_ASSERT(!intersects(Box{}, Box{}));
   // box with zero extent does
-  STATIC_ASSERT(intersects({{{0.0, 0.0, 0.0}}, {{0.0, 0.0, 0.0}}},
-                           {{{0.0, 0.0, 0.0}}, {{0.0, 0.0, 0.0}}}));
+  STATIC_ASSERT(intersects(Box{{{0.0, 0.0, 0.0}}, {{0.0, 0.0, 0.0}}},
+                           Box{{{0.0, 0.0, 0.0}}, {{0.0, 0.0, 0.0}}}));
```

Comment:
Make one commit for this with a message such as "Resolving ambiguity between intersects() Box and Ray overloads"